Copy editing (#198)
openforis/sepal-doc: docs/source/setup/gee.rst

Use Google Earth Engine (GEE) with SEPAL
==================================

GEE and SEPAL
-------------

SEPAL is a free, open-source cloud-computing platform that gives you access to geospatial data resources (either via the GEE data catalogue or user-provided data) and powerful computing resources to process these data to produce useful information.

SEPAL currently "resides" in the Amazon Web Services (AWS) ecosystem and makes use of AWS cloud computers, or instances, that can be used to process data using pre-programmed SEPAL applications or your own code (in R, Python or in the terminal). SEPAL is also closely linked to GEE, a Google-powered Earth-observation cloud-computing platform. SEPAL can access data stored in the GEE data catalog, use GEE for processing large datasets, and read and write data to GEE assets. A large part of SEPAL's existing code is based on the JavaScript GEE Application Programming Interface (API).  

> All SEPAL recipes are based on GEE and run scripts written by SEPAL team members to enable access to state-of-the-art processing algorithms without having to write code. In the SEPAL applications list, the applications tagged with the Google logo ("**G**") are also running with the Python GEE API and don't require you to use big instances to run complex operations. 

Reviewer: ```suggestion
All SEPAL recipes are based on GEE and run scripts written by SEPAL team members to enable access to state-of-the-art processing algorithms without having to write code. In the SEPAL applications list, the applications tagged with the Google logo (:icon:`fab fa-google`) are also running with the Python GEE API and don't require you to use big instances to run complex operations. 
```

since I wrote this page, I implemented an extention to use inline [font-awesome](https://fontawesome.com/v5/icons/google?s=brands) icons. let's use it here !

.. tip:

   See linking SEPAL and GEE accounts in the previous section for more information.

To get the best of SEPAL, we highly suggest that all users open their own GEE account and link it to SEPAL. 

In this section of SEPAL documentation, we will guide you through the full process of creating a GEE account and linking it to SEPAL. 

Create a GEE account
--------------------

Sign-up
^^^^^^^

Signing up for GEE is required to properly export images and data products from SEPAL as GEE assets or SEPAL applications using GEE. 

You will need to have a Google account to sign up. If you don't have one already, you can set one up here: https://accounts.google.com/servicelogin. 

To request a GEE account, please visit https://earthengine.google.com/new_signup/.

.. image:: ../_images/setup/register/gee_landing.png
   :alt: Request access to google earth engine.
   :align: center

> Once you have a GEE account, you can access the platform here: https://code.earthengine.google.com/.

Reviewer: ```suggestion
Once you have a GEE account, you can access the platform code editor here: https://code.earthengine.google.com/.
```

that's a very important distinction, GEE has 3 websites: the code editor, the platform itself, and the datasets.

.. image:: ../_images/setup/register/gee_code.png

>    :alt: GEE code editor

Reviewer: line 36 you added a "." at the end of the figure alt, but not here. When should I use one for figures legend ?

   :align: center

.. tip::

    When you first connect to the code editor, it is strongly advised to take the short tour provided in the interface. 

    .. image:: ../_images/setup/gee/editor_tour.png
        :alt: GEE code editor tour
        :align: center

.. tip::

    If you experience trouble while linking your Google account to GEE, we encourage you to contact the SEPAL team for support.
    
Initialize the **Home** folder
^^^^^^^^^^^^^^^^^^^^^^^^^^

> To use your GEE account in SEPAL, you need to  set up the **Home** folder, where all your assets (vectors, rasters, collections, mosaics, classifications, etc.) will be exported. Failing to set up this folder will prevent you from successfully executing export requests.

Reviewer: here I made a mistake, "vector" and "raster" are type of files that are normally known by GIS users. "collection" is a specific type of file in GEE. "mosaics" and "classification" are SEPAL recipes. In short I'm mixing different things at the same place. My guess is that instead we should focus on the file types in GEE only what do you think ?

.. image:: ../_images/setup/gee/create_home.png
    :alt: gee asset creation
    :align: center

This opens a pop-up menu to select the name of the folder, which can only be set once and never changed. If you're not satisfied with the suggested name, you can create your own (the only limitation is that you can not use spaces or special characters).

.. image:: ../_images/setup/gee/home_pop_up.png
    :alt: GEE popup for Home creation
    :align: center

When you return to your list of assets (**Zone 1** panel "**Assets**") you should see the name you provided as the first folder at the root of the asset tree. In this example, we used "galatheetest":

.. image:: ../_images/setup/gee/asset_tree.png
    :alt: asset tree

.. note:: 

    Now that you have initialized your GEE account, you can start the connection process between SEPAL and GEE.

Connection between GEE and SEPAL
--------------------------------

SEPAL can work without being connected to your GEE account, but you will miss numerous opportunities to interact with the platform. In this section, the connection procedure between GEE and SEPAL will be presented. 

Connection
^^^^^^^^^^

> The first step is to navigate to the SEPAL landing page (`sepal.io <https://sepal.io>`_) and sign in. 

Reviewer: ```suggestion
The first step is to navigate to the SEPAL landing page (`sepal.io <https://sepal.io>`__) and sign in. 
```

Click on :code:`Connect your Google account`. In the list provided, select the account associated with GEE: 

.. image:: ../_images/setup/gee/gee_credential.png

.. Note::

    You will see a message requesting your permission to authorize SEPAL's access to your Google Drive, as well as your **Asset** and **Home** folders to export recipes and module results.

After the connection process is complete, you will be redirected to the SEPAL website. If you click on the "User interface" button again, the pop-up window will display "Connected" in green on the upper-right side.

.. image:: ../_images/setup/gee/user_interface_connected.png
    :alt: SEPAL and GEE connected

Disconnection
^^^^^^^^^^^^^

If you want to change accounts or remove the link between your SEPAL and GEE accounts, you can disconnect SEPAL from GEE at any time. 

Return to the user interface and click on :code:`Google account`. You should see the following window: 

.. image:: ../_images/setup/gee/gee_connected.png
    :alt: gee connected 

Simply click on the :code:`Disconnect your Google account` button and your accounts will no longer be linked. 

Upload files to GEE 
-------------------

> When you use SEPAL recipes and modules, you'll be asked to use GEE assets to provide rasters or vectors to the core GEE-based algorithm. These assets can be either public assets that someone shared with you or assets that you created yourself and would like to use.

Reviewer: ```suggestion
When you use SEPAL recipes and modules, you'll be asked to use GEE assets to provide rasters (:code:`ee.Image`) or vectors (:code:`ee.FeatureCollection`) to the core GEE-based algorithm. These assets can be either public assets that someone shared with you or assets that you created yourself and would like to use.
```

putting the names that people already know (raster and vector) in correspondance with what they will use (Image and FeatureCollection). what do you think ?

.. tip::

>     For vector files, SEPAL provides an interface to upload them from your computer to the platform and eventually to GEE. This process allows you to deal with the full process directly from SEPAL without going to the GEE code interface. See the :doc:`../modules/dwn/import_to_gee` module documentation for more details.

Reviewer: ```suggestion
    For vector files, SEPAL provides an interface to upload them from your computer to the platform and eventually to GEE. This process allows you to deal with the full process directly from SEPAL without going to the GEE code interface. See the :doc:`../modules/dwn/vector_manager` module documentation for more details.
```

t he module actually changed name 6 month ago
